code/modules/reagents/chemistry/machinery/chem_heater.dm

```diff
@@ -33,11 +33,12 @@
 	///What state we're at in the tutorial
 	var/tutorial_state = 0
 
-/obj/machinery/chem_heater/Initialize()
+/obj/machinery/chem_heater/Initialize(mapload)
 	. = ..()
 	create_reagents(200, NO_REACT)//Lets save some calculations here
-	reagents.add_reagent(/datum/reagent/reaction_agent/basic_buffer, 20)
-	reagents.add_reagent(/datum/reagent/reaction_agent/acidic_buffer, 20)
+	if(mapload)
+		reagents.add_reagent(/datum/reagent/reaction_agent/basic_buffer, 20)
+		reagents.add_reagent(/datum/reagent/reaction_agent/acidic_buffer, 20)
```

Comment:
I feel like what's done for other machines in the codebase, is that you have a specific type that's used for mapping, something like `/obj/machinery/chem_heater/with_buffer`, that has these reagents in Initialize(). Meanwhile the regular construction/decon code produces boards that just make `/obj/machinery/chem_heater`, so you can't get more than one set of buffer reagents.

It has the same effect, but is more flexible in terms of spawning this stuff in later for example.

Reply:
I see! Just to make sure, that would require me to edit all the maps and replace them all, right? Or is there an easier way of implementing this?

Reply:
This would require you to edit the maps, yes, but you wouldn't need to use the map editor, this sort of type replacement is common enough that tools exist to aid you.

Reply:
Okay! I'll do just that. Give me a bit to finish up some other things and it should be done fairly quickly.

Reply:
Okay, how's that? Did I do it right?